core/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -292,4 +294,13 @@ public static void register(int id, DataTypeFactory dataTypeFactory) {
         }
         return streamer;
     }
+
+    public static DataType tryFindNotNullType(Iterable<? extends DataType> dataTypes) {
```

Comment:
I think a javadoc comment would be nice because it's not obvious what `try` implies here.